artifact/parser.cpp

```diff
@@ -108,17 +95,23 @@ ExpectedArtifact Parse(io::Reader &reader) {
 			parser_error::Code::ParseError,
 			"Got unexpected token " + tok.TypeToString() + " expected 'Header'"));
 	}
-	// auto header = v3::header::Parse(); // TODO (MEN-6178): Enable
+	auto expected_header = v3::header::Parse(*tok.value, config);
+	if (!expected_header) {
+		return expected::unexpected(parser_error::MakeError(
+			parser_error::Code::ParseError,
+			"Failed to parse the header: " + expected_header.error().message));
+	}
+	auto header = expected_header.value();
```

Comment:
Are we missing checksumming? (Or am I just missing where it is..?)

Reply:
Oh my, we're not. Fixing.